### PR TITLE
Configure TLS for local Pulp

### PR DESCRIPTION
**Pull request by m-bull**
- Use https version of Pulp container image if `pulp_enable_tls`.
- TLS cert/key location vars.
- Mount TLS cert/key files into container.

### CODE REVIEW

File: etc/kayobe/containers/pulp/pre.yml [hunk 1]
```diff
@@ -19,3 +19,24 @@
     dest: /opt/kayobe/containers/pulp/settings.py
     mode: 0644
   become: true
+
+- name: Configure TLS for local Pulp
+  when: pulp_enable_tls
```
Comment (markgoddard) [inline etc/kayobe/containers/pulp/pre.yml]:
```suggestion
  when: pulp_enable_tls | bool
```

File: etc/kayobe/pulp.yml [hunk 2]
```diff
@@ -1,10 +1,22 @@
 ---
 ###############################################################################
-# Local Pulp access credentials
+# Local Pulp server configuration
 
 # Base URL of the local Pulp service.
 # Default uses the seed node's IP on the admin network.
-pulp_url: "http://{{ admin_oc_net_name | net_ip(groups['seed'][0]) }}:80"
+pulp_url: "{{ 'https' if pulp_enable_tls else 'http' }}://{{ admin_oc_net_name | net_ip(groups['seed'][0]) }}:80"
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
pulp_url: "{{ 'https' if pulp_enable_tls | bool else 'http' }}://{{ admin_oc_net_name | net_ip(groups['seed'][0]) }}"
```

File: etc/kayobe/seed.yml [hunk 3]
```diff
@@ -96,7 +96,7 @@ seed_pulp_container:
     image: pulp/pulp
     pre: "{{ kayobe_config_path }}/containers/pulp/pre.yml"
     post: "{{ kayobe_config_path }}/containers/pulp/post.yml"
-    tag: "3.16"
+    tag: "{{ '3.16-https' if pulp_enable_tls else '3.16' }}"
```
Comment (markgoddard) [inline etc/kayobe/seed.yml]:
```suggestion
    tag: "{{ '3.16-https' if pulp_enable_tls | bool else '3.16' }}"
```

File: etc/kayobe/pulp.yml (hunk 2 above)
Comment (markgoddard) [reply etc/kayobe/pulp.yml]: I'm assuming it switches to 443 when using TLS? If not, should it?

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: etc/kayobe/pulp.yml (hunk 2 above)
Comment (m-bull) [reply etc/kayobe/pulp.yml]: Good one, thanks. Added `pulp_port` and a default setting.

**Comment by markgoddard** [comment]
Is it still a draft?

File: etc/kayobe/kolla.yml [hunk 4]
```diff
@@ -85,6 +85,11 @@ kolla_install_type: source
 # Docker namespace to use for Kolla images. Default is 'kolla'.
 kolla_docker_namespace: stackhpc
 
+# Whether docker should be configured to use an insecure registry for Kolla
+# images. Default is false, unless docker_registry_enabled is true and
+# docker_registry_enable_tls is false.
+kolla_docker_registry_insecure: "{{ 'https' not in stackhpc_repo_mirror_url }}"
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: I don't think this variable exists in kayobe in wallaby

Comment (m-bull) [reply etc/kayobe/kolla.yml]: Changed the target branch to Xena

**Review by markgoddard — APPROVED** [review]

**Pull request merged**